ariadne_codegen/contrib/shorter_results.py

```diff
@@ -0,0 +1,369 @@
+
+        super().__init__(schema, config_dict)
+
+    def generate_result_class(
+        self,
+        class_def: ast.ClassDef,
+        operation_definition: ExecutableDefinitionNode,
+        selection_set: SelectionSetNode,
+    ) -> ast.ClassDef:
+        """Store a map of all classes and its ast"""
+        self.class_dict[class_def.name] = class_def
+
+        return super().generate_result_class(
+            class_def, operation_definition, selection_set
+        )
+
+    def generate_client_module(self, module: ast.Module) -> ast.Module:
+        """
+        Update the generated client.
+
+        At this point we've the full ast for all classes used for the client so
+        instead of modifying class by class in `generate_client_method` we
+        modify each method from here. By doing so we ensure that we have a map
+        between all classes, including fragments, so we can traverse them to
+        figure out how many fields we have in total for each request.
+
+        If we only have a single field, including when expanding our fragments,
+        we can expand the single field and update the return type.
+        """
+        client_def = next(
+            filter(lambda o: isinstance(o, ast.ClassDef), module.body), None
+        )
+        if not client_def:
+            return super().generate_client_module(module)
+
+        for method_def in [
+            m
+            for m in client_def.body
+            if isinstance(m, (ast.FunctionDef, ast.AsyncFunctionDef))
+        ]:
+            self._modify_method_def(method_def)
+
+        if len(self.extended_imports) == 0:
+            return super().generate_client_module(module)
+
+        for stmt in module.body:
+            # We know we've already imported the wrapping type so we're only
+            # interested in `ImportFrom`.
+            if not isinstance(stmt, ast.ImportFrom):
+                continue
+
+            # The module we import from is always the same as the method we
+            # modified when changing the return type.
+            if stmt.module not in self.extended_imports:
+                continue
+
+            # Add all additional imports discovered when generating the client
+            # method.
+            for additional_import in self.extended_imports[stmt.module]:
+                stmt.names.append(ast.alias(name=additional_import))
+
+        return super().generate_client_module(module)
+
+    def _modify_method_def(
+        self, method_def: Union[ast.FunctionDef, ast.AsyncFunctionDef]
+    ) -> Union[ast.FunctionDef, ast.AsyncFunctionDef]:
+        """
+        Change the method generated in the client to call the inner field and
+        change the return type. We do this here instead of
+        `generate_client_method` to ensure we have a map between all our classes
+        when updating the method def.
+        """
+        if len(method_def.body) < 1:
+            return super().generate_client_method(method_def)
```

Comment:
With current approach `generate_client_method` is no longer used, so we should switch to something else as fallback. We don't use output of this method so imo we could just use `return None` instead.

Reply:
Ah, returning `None` in all the paths from non inherited classes now instead!